hazelcast/src/main/java/com/hazelcast/sql/impl/schema/map/MapResolverPlugin.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.schema.map;
+
+import com.hazelcast.map.IMap;
+import com.hazelcast.replicatedmap.ReplicatedMap;
+
+/**
+ * Plugin mechanism, allowing to perform additional analysis of
+ * {@link IMap}/{@link ReplicatedMap} keys & values on sample based schema resolution.
+ * <p/>
+ * Used by Jet.
+ */
+@FunctionalInterface
+public interface MapResolverPlugin {
+
+    /**
+     * Return an appendix that is attached to the resolved {@link AbstractMapTable}.
+     */
+    Object resolve(Object object, boolean key);
```

Comment:
Also minor comment.

In the resolver's code, you either pass here Class of ClassDefinition as the first argument. It is ok from IMDG perspective, but are you ok with a necessity of casts inside Jet implementation? 

If you leave the interface as is, I would propose adding JavaDoc about expected arguments.

Reply:
I had an impression that it was the proposed solution. Anyway, what about this one then?
```
interface JetMapMetadataResolver {
    Object resolveClass(Class<?> clazz, boolean key);

    Object resolvePortable(ClassDeficition clazz, boolean key);
}
```
Additional methods could be added if needed (i.e. Json).

Reply:
They both OK from the IMDG perspective. The question is - what is better for Jet. 
The interface you proposed in the comment looks good to me - it is clean and simple.